File: idea-plugin/src/main/java/com/zzf/rikki/idea/completion/CompletionSettings.java
package com.zzf.rikki.idea.completion;

public final class CompletionSettings {
    private static boolean enabled = Boolean.parseBoolean(System.getProperty("rikki.completion.enabled", "true"));
    private static String endpoint = System.getProperty("rikki.completion.endpoint", resolveDefaultEndpoint());

    private CompletionSettings() {
    }

    public static boolean isEnabled() {
        return enabled;
    }

    public static void setEnabled(boolean value) {
        enabled = value;
    }

    public static String getEndpoint() {
        return endpoint;
    }

    public static void setEndpoint(String value) {
        endpoint = value == null ? "" : value;
    }

    private static String resolveDefaultEndpoint() {
        String base = System.getProperty("rikki.endpoint", "http://localhost:18080/api/agent/chat");
        if (base.endsWith("/chat/stream")) {
            return base.substring(0, base.length() - "/stream".length()).replaceAll("/chat$", "") + "/complete";
        }
        if (base.endsWith("/chat")) {
            return base.substring(0, base.length() - "/chat".length()) + "/complete";
        }
        return base + "/complete";
    }
}

File: idea-plugin/src/main/kotlin/com/zzf/rikki/idea/completion/CompletionSettings.kt
package com.zzf.rikki.idea.completion

/**
 * Global settings for inline completion.
 * Values can be overridden via JVM system properties for easy testing:
 *   -Dcodeagent.completion.enabled=true
 *   -Dcodeagent.completion.endpoint=http://localhost:18080/api/agent/complete
 */
object CompletionSettings {

    /** Master on/off switch. */
    var isEnabled: Boolean
        get() = _enabled
        set(value) { _enabled = value }

    /** Backend FIM endpoint URL. */
    var endpoint: String
        get() = _endpoint
        set(value) { _endpoint = value }

    // ── Backing fields with system-property override support ──────────────────

    private var _enabled: Boolean =
        System.getProperty("rikki.completion.enabled", "true").toBoolean()

    private var _endpoint: String =
        System.getProperty(
            "rikki.completion.endpoint",
            resolveDefaultEndpoint()
        )

    private fun resolveDefaultEndpoint(): String {
        val base = System.getProperty("rikki.endpoint", "http://localhost:18080/api/agent/chat")
        // Derive /complete endpoint from the existing chat endpoint
        return when {
            base.endsWith("/chat/stream") ->
                base.removeSuffix("/stream").removeSuffix("/chat") + "/complete"
            base.endsWith("/chat") ->
                base.removeSuffix("/chat") + "/complete"
            else -> "$base/complete"
        }
    }
}
